### PR TITLE
Support compressed TIFFs

### PR DESCRIPTION
**Pull request by tpendragon**

### CODE REVIEW

File: app/derivative-services/jp2_derivative_service.rb [hunk 1]
```diff
@@ -114,7 +114,30 @@ def use
...
+      rescue MiniMagick::Error
+        filename
+      end
+  end
+
+  # Remove compression from given TIFF file, just in case.
+  def cleaned_file
+    @cleaned_file ||=
+      begin
+        t = Tempfile.new
```
Comment (escowles) [inline app/derivative-services/jp2_derivative_service.rb]: Do we need to explicitly clean up this tempfile after the derivative processing happens?

Comment (tpendragon) [reply app/derivative-services/jp2_derivative_service.rb]: No, Tempfile means it should get deleted after the process closes.

File: app/derivative-services/jp2_derivative_service.rb [hunk 2]
```diff
@@ -69,7 +69,7 @@ def run_derivatives
 
   def run_tiff_derivatives
     Hydra::Derivatives::Jpeg2kImageDerivatives.create(
-      filename,
+      clean_filename,
```
[the remark this answers is not in the document]
Comment (tpendragon) [reply app/derivative-services/jp2_derivative_service.rb]: Yeah, I can change that on Monday.

[the remark this answers is not in the document]
Comment (tpendragon) [reply app/derivative-services/jp2_derivative_service.rb]: @escowles Fixed.

File: app/derivative-services/jp2_derivative_service.rb [hunk 3]
```diff
@@ -114,7 +120,28 @@ def use
...
+  end
+
+  # Remove compression from given TIFF file, just in case.
+  def cleaned_file
+    @cleaned_file ||=
+      begin
+        t = Tempfile.new
+        MiniMagick::Tool::Convert.new do |convert|
+          convert << file_object.disk_path.to_s
+          convert.compress.+
```
Comment (hackmastera) [inline app/derivative-services/jp2_derivative_service.rb]: what does this do? Is it some kind of DSL for the command line option?

File: app/derivative-services/jp2_derivative_service.rb [hunk 4]
```diff
@@ -114,7 +120,28 @@ def use
   end
 
   def filename
-    return Pathname.new(file_object.io.path) if file_object.io.respond_to?(:path) && File.exist?(file_object.io.path)
+    Pathname.new(file_object.disk_path)
```
Comment (hackmastera) [inline app/derivative-services/jp2_derivative_service.rb]: I wonder why these error checks were in here. your change is definitely an improvement, but it makes me wonder if there were errors? Not sure how there could be an io without a path.

File: app/derivative-services/jp2_derivative_service.rb (hunk 3 above)
Comment (tpendragon) [reply app/derivative-services/jp2_derivative_service.rb]: Yeah. Makes the command line `compress+`

File: app/derivative-services/jp2_derivative_service.rb (hunk 4 above)
Comment (tpendragon) [reply app/derivative-services/jp2_derivative_service.rb]: This code is from the Valkyrie MVP and existed before `disk_path` was a required property in the Valkyrie API.

**Comment by tpendragon** [comment]
> How did you get around the tika stub problem?

I didn't - it's a tiff, so the stubbed characterization is fine.

**Pull request merged**